src/value.c

```diff
@@ -76,11 +82,28 @@ SIValue SI_TransferStringVal(char *s) {
 	};
 }
 
+// deep clone an array of SIValues
+SIValue SI_CloneArrayVal(SIValue *array) {
+	size_t arrayLen = array_len(array);
+	SIValue *clone = array_new(SIValue, arrayLen);
+	for(int i = 0; i < arrayLen; i++) {
+		SIValue clonedValue = SI_Clone(array[i]);
+		array_append(clone, clonedValue);
+	}
+	SIValue value;
+	value.array = clone;
+	value.type = T_ARRAY;
```

Comment:
I think calling `SI_Array(clone)` to initialize would be preferable here. If not, set `value.allocation = M_VOLATILE`.

Reply:
changed to `return SI_Array(clone);`